overlord/overlord.go

```diff
@@ -167,6 +165,50 @@ func (o *Overlord) Stop() error {
 	return err1
 }
 
+// Settle runs first a state engine Ensure and then wait for activities to settle.
+// That's done by waiting for all managers activities to settle while
+// making sure no immediate further Ensure is scheduled. Chiefly for tests.
+// Cannot be used in conjuction with Loop.
+func (o *Overlord) Settle() error {
+	func() {
```

Comment:
Just curious: I assume this anonymous function is used for the nice  `Lock(); defer Unlock() ` - or is there another reason?